common/env.go

```diff
@@ -16,6 +18,26 @@ var (
 	commonClient *DatabricksClient
 )
 
+// GetAzureEnvironment returns azure environment from giving environment string
+func GetAzureEnvironment(environment string) (azure.Environment, error) {
```

Comment:
1) I'd advise to move this function back to azure_auth.go, as this file has broader scope. E.g. provider.go works a lot with env, but it doesn't mean it should be defined here :)

2) merge these two functions. GetAzureEnvironment is supposed to return error, but it does never return it. And the other function logs the error instead of returning it. Priority for error is to propagate it up (as Go doesn't have exceptions...), not just logging.